src/sc/ContractParam.js

```diff
@@ -62,13 +62,29 @@ class ContractParam {
    * Creates a ByteArray ContractParam.
    * @param {any} value
    * @param {string} format - The format that this value represents. Different formats are parsed differently.
+   * @param {any} args - Additional arguments such as decimal precision
    * @return {ContractParam}
    */
-  static byteArray (value, format) {
+  static byteArray (value, format, ...args) {
     if (format) format = format.toLowerCase()
     if (format === 'address') {
       return new ContractParam('ByteArray', reverseHex(getScriptHashFromAddress(value)))
     } else if (format === 'fixed8') {
+      var decimals = 8
+      if (args.length === 1) {
+        decimals = args[0]
+      }
+      if (!isFinite(value)) throw new Error(`Input should be number!`)
+      var e = 1
+      var p = 0
+      while (Math.round(value * e) / e !== value) {
+        e *= 10
+        p++
+      }

```

Comment:
A `while` loop looks very dangerous here. I would suggest just using the Fixed8 class to determine if your precision is exceeded.
```js
const fixed8Value = new Fixed8(value)
const adjustedValue = fixed8Value.times(Math.pow(10, 8 - decimals))
const modValue = adjustValue.mod(1)
if (!modValue.isZero()) throw new Error(`wrong precision`)
return adjustedValue
```